packages/flutter_adaptive_scaffold/lib/src/adaptive_scaffold.dart

```diff
@@ -235,10 +235,12 @@ class AdaptiveScaffold extends StatefulWidget {
   /// Public helper method to be used for creating a [NavigationRailDestination] from
   /// a [NavigationDestination].
   static NavigationRailDestination toRailDestination(
-      NavigationDestination destination) {
+    NavigationDestination destination,
+  ) {
     return NavigationRailDestination(
       label: Text(destination.label),
       icon: destination.icon,
+      selectedIcon: destination.selectedIcon,
```

Comment:
In the `test/adaptive_scaffold_test.dart` file, you can create a new test to find this icon to make sure that it is being displayed. You might have to have the test tap on the button to select it.  To find the icon's widget, you can use `find.byIcon()`, giving it the selected icon you designated in the test.

Reply:
Test cases added for determining selected Icon availability in UI if any tap gesture happens to navigation rail.